rust/lance/benches/take.rs

```diff
@@ -67,27 +97,135 @@ fn bench_take_for_version(
     }
 }
 
-fn bench_random_take(c: &mut Criterion) {
+fn bench_random_take_with_file_reader(c: &mut Criterion) {
+    // default tokio runtime
     let rt = tokio::runtime::Runtime::new().unwrap();
     let num_batches = 1024;
+    let file_size = num_batches * BATCH_SIZE as i32 + 1;
 
-    for file_size in [1024 * 1024, 1024] {
-        bench_take_for_version(
-            c,
-            &rt,
-            file_size,
-            num_batches,
-            LanceFileVersion::V2_0,
-            "V2_0",
-        );
-        bench_take_for_version(
-            c,
-            &rt,
-            file_size,
-            num_batches,
-            LanceFileVersion::V2_1,
-            "V2_1",
-        );
+    file_reader_take(c, &rt, file_size, num_batches, LanceFileVersion::V2_0, "V2_0");
+    file_reader_take(c, &rt, file_size, num_batches, LanceFileVersion::V2_1, "V2_1");
+}
+
+fn file_reader_take(
+    c: &mut Criterion,
+    rt: &tokio::runtime::Runtime,
+    file_size: i32,
+    num_batches: i32,
+    version: LanceFileVersion,
+    version_name: &str,
+) {
+    let (dataset, file_path) = rt.block_on(async {
+        // Make sure there is only one fragment.
+        let dataset = create_dataset("memory://test.lance", version, num_batches, file_size).await;
+
+        assert_eq!(dataset.get_fragments().len(), 1);
+        let fragments = dataset.get_fragments();
+        let fragment = fragments.first().unwrap();
+        assert_eq!(fragment.num_data_files(), 1);
+        let file = fragment.metadata().files.get(0).unwrap();
+        let file_path = dataset.data_dir().child(file.path.as_str());
+
+        (dataset, file_path)
+    });
+
+    // Bench random take.
+    for num_rows in [1, 10, 100, 1000] {
+        c.bench_function(&format!(
+            "{version_name} Random Take FileReader({file_size} file size, {num_batches} batches, {num_rows} rows per take)"
+        ), |b| {
+            b.to_async(rt).iter(|| async {
+                let file_reader = create_file_reader(&dataset, &file_path).await;
+
+                let rows = gen_ranges(num_batches as u64 * BATCH_SIZE, file_size as u64, num_rows);
+                for r in rows {
```

Comment:
this should not iterate through each row to take, but use `ReadBatchParams::Indices`